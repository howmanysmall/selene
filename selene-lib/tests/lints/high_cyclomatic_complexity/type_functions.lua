-- Type functions should not contribute to cyclomatic complexity
-- since they are compile-time constructs

type function simple_type_function()
    return {}
end

export type function exported_type_function()
    return {}
end

-- Regular function should still contribute to complexity (this should warn)
local function regular_function()
    if true then
        if true then
            if true then
                if true then
                    if true then
                        if true then
                            if true then
                                if true then
                                    if true then
                                        if true then
                                            print("too complex")
                                        end
                                    end
                                end
                            end
                        end
                    end
                end
            end
        end
    end
end
