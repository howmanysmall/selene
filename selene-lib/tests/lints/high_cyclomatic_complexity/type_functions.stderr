error[high_cyclomatic_complexity]: this function has a cyclomatic complexity of 11
   ┌── type_functions.lua:11:1 ───
   │
11 │ local function regular_function()
   │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │
   = note: consider refactoring this function
